src/altinn-app-frontend/src/features/form/containers/RepeatingGroupTable.tsx

```diff
@@ -487,16 +527,59 @@ export function RepeatingGroupTable({
+                            })}
                           >
-                            <IconButton
-                              className={classes.deleteButton}
-                              disabled={deleting}
-                              onClick={removeClicked(index)}
-                              aria-label={`${deleteButtonText}-${firstCellData}`}
+                            <PopoverPanel
+                              variant={PanelVariant.Warning}
+                              side={'left'}
+                              open={popoverPanelIndex == index && popoverOpen}
+                              onOpenChange={setPopoverOpen}
+                              showIcon={false}
+                              forceMobileLayout={true}
+                              trigger={
+                                <IconButton
+                                  className={classes.deleteButton}
+                                  disabled={deleting}
+                                  onClick={handleDeleteClick(index)}
+                                  aria-label={`${deleteButtonText}-${firstCellData}`}
+                                >
+                                  <i className='ai ai-trash' />
+                                  {deleteButtonText}
+                                </IconButton>
+                              }
                             >
-                              <i className='ai ai-trash' />
-                              {deleteButtonText}
-                            </IconButton>
+                              <div>
+                                {getLanguageFromKey(
+                                  'group.row_popover_delete_message',
+                                  language,
+                                )}
+                              </div>
+                              <div
+                                className={cn(classes.popoverButtonContainer)}
+                              >
+                                <Button
+                                  variant={ButtonVariant.Cancel}
+                                  onClick={handlePopoverDeleteClick(index)}
+                                >
+                                  {getLanguageFromKey(
+                                    'group.row_popover_delete_button_confirm',
+                                    language,
+                                  )}
+                                </Button>
+                                <Button
+                                  variant={ButtonVariant.Secondary}
+                                  onClick={() => onOpenChange(index)}
+                                >
+                                  {getLanguageFromKey(
+                                    'general.cancel',
+                                    language,
+                                  )}
+                                </Button>
+                              </div>
+                            </PopoverPanel>
```

Comment:
This component is getting quite big - perhaps this should be extracted to smaller parts?